src/hoc/withTheme.js

```diff
@@ -32,17 +33,19 @@ const wrapWithTheme = (Component: ReactClass<any>) => {
 
     componentWillMount() {
       const styledContext = this.context[CHANNEL_NEXT]
-      if (styledContext === undefined) {
+      if (!styledContext && !this.constructor.defaultProps) {
         // eslint-disable-next-line no-console
-        console.error('[withTheme] Please use ThemeProvider to be able to use withTheme')
-        return
-      }
-
-
-      const { subscribe } = styledContext
-      this.unsubscribeId = subscribe(theme => {
+        console.warn('[withTheme] You are not using a ThemeProvider nor passing a theme in defaultProps')
+      } else if (!styledContext && this.contructor.defaultProps) {
+        const theme = determineTheme(this.props, undefined, this.constructor.defaultProps)
```

Comment:
There’s no such logic in componentWillReceiveProps which will allow the theme prop to update